streams/src/test/java/org/apache/kafka/streams/TopologyTest.java

```diff
@@ -276,15 +276,13 @@ public void shouldThrowOnUnassignedStateStoreAccess() throws Exception {
 
         try {
             new ProcessorTopologyTestDriver(streamsConfig, topology.internalTopologyBuilder);
+            fail("Should have thrown StreamsException");
         } catch (final StreamsException e) {
-            final Throwable cause = e.getCause();
-            if (cause != null
-                && cause instanceof TopologyBuilderException
-                && cause.getMessage().equals("Invalid topology building: Processor " + badNodeName + " has no access to StateStore " + LocalMockProcessorSupplier.STORE_NAME)) {
-                throw (TopologyBuilderException) cause;
-            } else {
-                throw new RuntimeException("Did expect different exception. Did catch:", e);
-            }
+            final String error = e.getCause().toString();
+            final String expectedMessage = "Processor " + badNodeName + " has no access to StateStore " +
+                    LocalMockProcessorSupplier.STORE_NAME;
+            
+            assertTrue(error.contains(expectedMessage));
```

Comment:
Why not `assertEquals` ?

Reply:
Its possible.  I'll have to add the fully qualified reference to `StreamsException` because that's part of the error message.